### ISSUE TITLE
logical source reference from the core part of the spec

### ISSUE THREAD

**Issue by andimou** [kg-construct/rml-fno-spec#25]
The core description has references to some input data but if we split the logical source and core part of the spec, how should we handle the references of the examples at the core part?

Should we introduce an abstract running example? or should we still have examples for all exemplary data sources?

**Comment by pmaria** [comment]
I don't think we should split the logical source part completely from the core spec. I think you would need at least an abstract / general descpription of what a logical source is and how it works, irrespective of specific reference formulations, to be able to describe the core workings. Then there can be one (or more) separate spec(s) for concrete reference formulations.

**Comment by andimou** [comment]
so you think that the RML-core should include both the description of the Logical Source and the rest of Triples Map?

Because in past discussions, we were thinking of having a separate spec for the Logical Source accompanied by different notes for different reference formulations.

**Comment by pmaria** [comment]
Well I guess it is question of preference. But I'd say that the logical source is core to the understanding of RML. So I think it would be good to describe it in the core spec. I think it will be hard to describe how a Triples Map generates results without, at least abstractly, describing the logical source.

**Comment by andimou** [comment]
I think for this we discussed two alternatives:

- before the summer holidays, we discussed the option of having these intermediate key-value combinations considered by the core spec and then make this concrete at the other parts of the spec
- after the summer holidays, we discussed keeping it as in the case of R2RML being the default behaviour but without the constraints, so we have room to specify different at the other specs.
For instance, default spec would say that the subject can be an IRI (without say only IRI, ie it doesn't exclude other options) and then the star part of the spec can say that the sucbject can also be a star term.

**Comment by dachafra** [comment]
@DylanVanAssche I think it's clear that we are going to have a LogicalSource spec, but... what are the current changes we need to do over the core spec to be adapted and self-contained? any idea? I also agree that at least, an abstract way for describing the logical source in the core is needed.

**Comment by DylanVanAssche** [comment]
> what are the current changes we need to do over the core spec to be adapted and self-contained?

- I would keep `rml:logicalSource` and `rml:logicalTarget` in the Core Spec
- Move all Logical Source stuff (class, properties, etc.) to the Logical Source Spec.

> I also agree that at least, an abstract way for describing the logical source in the core is needed.

I think by keeping `rml:logicalSource` and `rml:logicalTarget` we have enough. You can allow any Object here so you don't depend on the Logical Source / Target spec, but they can be used here to have a straightforward integration.

**Comment by DylanVanAssche** [comment]
We did what I presented in https://github.com/kg-construct/rml-core/issues/25#issuecomment-1066772897
Let's close this issue since we have a proposal that is implemented.
